mir-module/src/main/resources/xsl/mods2oaiheader.xsl

```diff
@@ -28,6 +37,49 @@
     <xsl:value-of select="substring-before(service/servdates/servdate[@type='modifydate'],'T')" />
   </datestamp>
 
+  <!-- check for the following sets: doc-type,open_access,openaire,driver,ec_fundedresources,GENRE,ddc -->
+  <setSpec><xsl:value-of select="concat('doc-type:', substring-after(metadata/def.modsContainer/modsContainer/mods:mods/mods:classification[@generator='mir_genres2diniPublType-mycore']/@valueURI,'#'))" /></setSpec>
+  <setSpec><xsl:value-of select="concat('GENRE:', substring-after(metadata/def.modsContainer/modsContainer/mods:mods/mods:genre/@valueURI,'#'))" /></setSpec>
+
+  <xsl:choose>
+    <xsl:when test="metadata/def.modsContainer/modsContainer/mods:mods/mods:classification[@authority='ddc']">
```

Comment:
shouldn't we ask our classification system for this? It would also handle parent categories for doc-type and such. Maybe it is good to move classification-typed sets to a common template with parameters: setSpec-prefix, categoryNodes